server/rest/dataset.py

```diff
@@ -530,3 +532,35 @@ def createTweetGeocodes(self, item, params):
         .param('id', 'The Item ID', paramType='path')
         .errorResponse('ID was invalid.')
         .errorResponse('Write permission denied on the Item.', 403))
+
+    @access.public
+    @loadmodel(map={'wmsSourceId': 'wmsSource'}, model='item',
+               level=AccessType.READ)
+    def createWmsDataset(self, wmsSource, params):
```

Comment:
@aashish24  This is an API endpoint definition, and is so far the only functionality of the wmsDataset on the server-side.  I'm not sure what it would gain us to do what you suggest.

I can see having most of the code in this method be in a common method or superclass, but beyond that I'm not sure.

Maybe you can post some specifics or create an experimental PR with some suggested changes?